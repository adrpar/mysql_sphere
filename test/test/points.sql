SET sphere_outprec=8;

-- operators
SELECT sequal( spoint( '(0, 90d)' )  , spoint( '(0, 90d)' ) );
SELECT sequal( spoint( '(0, 90d)' )  , spoint( '(0,-90d)' ) );
SELECT sequal( spoint( '(0,-90d)' )  , spoint( '(0,-90d)' ) );
SELECT snotequal( spoint( '(0, 90d)' ) , spoint( '(0, 90d)' ) );
SELECT snotequal( spoint( '(0, 90d)' ) , spoint( '(0,-90d)' ) );
SELECT snotequal( spoint( '(0,-90d)' ) , spoint( '(0,-90d)' ) );
SELECT sequal( spoint( '(0d,  0)' )  , spoint( '(360d,0)' ) );

-- I/O test --

-- Output ---

SET sphere_outmode="DEG";

SELECT sstr( spoint( '( 1h 2m 30s , +1d 2m 30s)' ) );

SELECT sstr( spoint( '( 0h 2m 30s , +0d 2m 30s)' ) );

SELECT sstr( spoint( '( 1h 3m 30s , -1d 3m 3.6s)' ) );

SELECT sstr( spoint( '( 0h 3m 30s , -0d 3m 3.6s)' ) );

SELECT sstr( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) );

SELECT sstr( spoint( '( 0h 2m 30s ,-90d 0m 0s)' ) );

SET sphere_outmode="DMS";

SELECT sstr( spoint( '( 1h 2m 30s , +1d 2m 30s)' ) );

SELECT sstr( spoint( '( 0h 2m 30s , +0d 2m 30s)' ) );

SELECT sstr( spoint( '( 1h 2m 30s , -1d 2m 30s)' ) );

SELECT sstr( spoint( '( 0h 2m 30s , -0d 2m 30s)' ) );

SELECT sstr( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) );

SELECT sstr( spoint( '( 0h 2m 30s ,-90d 0m 0s)' ) );

SET sphere_outmode="HMS";

SELECT sstr( spoint( '( 1h 2m 30s , +1d 2m 30s)' ) );

SELECT sstr( spoint( '( 0h 2m 30s , +0d 2m 30s)' ) );

SELECT sstr( spoint( '( 1h 2m 30s , -1d 2m 30s)' ) );

SELECT sstr( spoint( '( 0h 2m 30s , -0d 2m 30s)' ) );

SELECT sstr( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) );

SELECT sstr( spoint( '( 0h 2m 30s ,-90d 0m 0s)' ) );

SET sphere_outmode="RAD";

SELECT sstr( spoint( '( 0h 0m 0s , 0d 0m 0s)' ) );

-- "incorrect dec. values"

SET sphere_outmode="DEG";

SELECT sstr( spoint( '( 0h 2m 30s , 95d 0m 0s)' ) );

SELECT sstr( spoint( '( 24h 2m 30s , 5d 0m 0s)' ) );

SELECT sstr( spoint( '( -0h 2m 30s , -5d 0m 0s)' ) );

SELECT sstr( spoint( '( 0h 2m 30s , -95d 0m 0s)' ) );

-- Input --

SELECT sstr( spoint( '( 0.625d , -0.04166666667d)' ) );

SELECT sstr( spoint( '(0.0109083078249646 , -0.000727220521664407)' ) );

-- functions for point --------------

-- sstr( spoint(float8, float8 ))

SELECT sstr( spoint(0.0109083078249646 , -0.000727220521664407) );

SET sphere_outmode="RAD";

SELECT sstr( spoint(7.28318530717958623 , 0.00) );

SELECT sstr( spoint(0.0 , 2.141592653589793116) );

-- sdist(spoint, spoint)

SELECT sdist(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 0m 30s , 0d 0m 0s)' ) );

SELECT sdist(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 2m 30s , 10d 0m 0s)' ) );

SELECT sdist(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 2m 30s , -10d 0m 0s)' ) );

SELECT sdist(spoint( '( 0h 2m 30s , 95d 0m 0s)' ) , spoint( '( 12h 2m 30s , 85d 0m 0s)' ) );

SELECT sdist(spoint( '( 24h 2m 30s , 10d 0m 0s)' ) , spoint( '( 0h 2m 30s , -10d 0m 0s)' ) );

SELECT sdist(spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) );

SELECT sdist(spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

-- spoint_long(spoint)

SELECT spoint_long(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) );

SELECT spoint_long(spoint( '( 0h 2m 30s ,95d 0m 0s)' ) );

SELECT spoint_long(spoint( '( 0h 2m 30s ,85d 0m 0s)' ) );

SELECT spoint_long(spoint( '( 0h 2m 30s ,-95d 0m 0s)' ) );

SELECT spoint_long(spoint( '( 0h 2m 30s ,-85d 0m 0s)' ) );

SELECT spoint_long(spoint( '( 0h 2m 30s ,90d 0m 0s)' ) );

SELECT spoint_long(spoint( '( 0h 2m 30s ,-90d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s , 0d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s ,95d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s ,85d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s ,-95d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s ,-85d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s ,90d 0m 0s)' ) );

SELECT spoint_long(spoint( '(24h 2m 30s ,-90d 0m 0s)' ) );

-- spoint_lat(spoint)

SELECT spoint_lat(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) );

SELECT spoint_lat(spoint( '( 0h 2m 30s ,95d 0m 0s)' ) );

SELECT spoint_lat(spoint( '( 0h 2m 30s ,85d 0m 0s)' ) );

SELECT spoint_lat(spoint( '( 0h 2m 30s ,-95d 0m 0s)' ) );

SELECT spoint_lat(spoint( '( 0h 2m 30s ,-85d 0m 0s)' ) );

SELECT spoint_lat(spoint( '( 0h 2m 30s ,90d 0m 0s)' ) );

SELECT spoint_lat(spoint( '( 0h 2m 30s ,-90d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s , 0d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s ,95d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s ,85d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s ,-95d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s ,-85d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s ,90d 0m 0s)' ) );

SELECT spoint_lat(spoint( '(24h 2m 30s ,-90d 0m 0s)' ) );

-- xyz(spoint)

SELECT spoint_x(spoint( '(0d,-90d)' ) );

SELECT spoint_y(spoint( '(0d,-90d)' ) );

SELECT spoint_z(spoint( '(0d,-90d)' ) );

-- operators for points -------------

-- opsequal( erator ---------------------- )-

SELECT sequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) );

SELECT sequal( spoint( '( 12h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) );

SELECT sequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 24h 2m 30s , 90d 0m 0s)' ) );

SELECT sequal( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT sequal( spoint( '( 12h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT sequal( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 24h 2m 30s , -90d 0m 0s)' ) );

SELECT sequal( spoint( '( 0h 2m 30s , 95d 0m 0s)' ) , spoint( '( 12h 2m 30s , 85d 0m 0s)' ) );

SELECT sequal( spoint( '( 24h 2m 30s , 10d 0m 0s)' ) , spoint( '( 0h 2m 30s , 10d 0m 0s)' ) );

SELECT sequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT sequal( spoint( '( 12h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT sequal( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 12h 2m 30s , 45d 0m 0s)' ) );

-- <> operator -----------------------

SELECT snotequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 12h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 24h 2m 30s , 90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 12h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 24h 2m 30s , -90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 0h 2m 30s , 95d 0m 0s)' ) , spoint( '( 12h 2m 30s , 85d 0m 0s)' ) );

SELECT snotequal( spoint( '( 24h 2m 30s , 10d 0m 0s)' ) , spoint( '( 0h 2m 30s , 10d 0m 0s)' ) );

SELECT snotequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 12h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

SELECT snotequal( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 12h 2m 30s , 45d 0m 0s)' ) );

-- <-> operator ---------------------

SELECT sdist( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 0m 30s , 0d 0m 0s)' ) );

SELECT sdist( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 2m 30s , 10d 0m 0s)' ) );

SELECT sdist( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 2m 30s , -10d 0m 0s)' ) );

SELECT sdist( spoint( '( 0h 2m 30s , 95d 0m 0s)' ) , spoint( '( 12h 2m 30s , 85d 0m 0s)' ) );

SELECT sdist( spoint( '( 24h 2m 30s , 10d 0m 0s)' ) , spoint( '( 0h 2m 30s , -10d 0m 0s)' ) );

SELECT sdist( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) );

SELECT sdist( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) );

