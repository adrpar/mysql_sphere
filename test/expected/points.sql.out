sequal( spoint( '(0, 90d)' )  , spoint( '(0, 90d)' ) )
1
sequal( spoint( '(0, 90d)' )  , spoint( '(0,-90d)' ) )
0
sequal( spoint( '(0,-90d)' )  , spoint( '(0,-90d)' ) )
1
snotequal( spoint( '(0, 90d)' ) , spoint( '(0, 90d)' ) )
0
snotequal( spoint( '(0, 90d)' ) , spoint( '(0,-90d)' ) )
1
snotequal( spoint( '(0,-90d)' ) , spoint( '(0,-90d)' ) )
0
sequal( spoint( '(0d,  0)' )  , spoint( '(360d,0)' ) )
1
sstr( spoint( '( 1h 2m 30s , +1d 2m 30s)' ) )
(15.625d , 1.0416667d)
sstr( spoint( '( 0h 2m 30s , +0d 2m 30s)' ) )
(0.625d , 0.041666667d)
sstr( spoint( '( 1h 3m 30s , -1d 3m 3.6s)' ) )
(15.875d , -1.051d)
sstr( spoint( '( 0h 3m 30s , -0d 3m 3.6s)' ) )
(0.875d , -0.051d)
sstr( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) )
(0.625d , 90d)
sstr( spoint( '( 0h 2m 30s ,-90d 0m 0s)' ) )
(0.625d , -90d)
sstr( spoint( '( 1h 2m 30s , +1d 2m 30s)' ) )
( 15d 37m 30s , + 1d  2m 30s)
sstr( spoint( '( 0h 2m 30s , +0d 2m 30s)' ) )
(  0d 37m 30s , + 0d  2m 30s)
sstr( spoint( '( 1h 2m 30s , -1d 2m 30s)' ) )
( 15d 37m 30s , - 1d  2m 30s)
sstr( spoint( '( 0h 2m 30s , -0d 2m 30s)' ) )
(  0d 37m 30s , - 0d  2m 30s)
sstr( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) )
(  0d 37m 30s , +90d  0m 0s)
sstr( spoint( '( 0h 2m 30s ,-90d 0m 0s)' ) )
(  0d 37m 30s , -90d  0m 0s)
sstr( spoint( '( 1h 2m 30s , +1d 2m 30s)' ) )
(  1h  2m 30s , + 1d  2m 30s)
sstr( spoint( '( 0h 2m 30s , +0d 2m 30s)' ) )
(  0h  2m 30s , + 0d  2m 30s)
sstr( spoint( '( 1h 2m 30s , -1d 2m 30s)' ) )
(  1h  2m 30s , - 1d  2m 30s)
sstr( spoint( '( 0h 2m 30s , -0d 2m 30s)' ) )
(  0h  2m 30s , - 0d  2m 30s)
sstr( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) )
(  0h  2m 30s , +90d  0m 0s)
sstr( spoint( '( 0h 2m 30s ,-90d 0m 0s)' ) )
(  0h  2m 30s , -90d  0m 0s)
sstr( spoint( '( 0h 0m 0s , 0d 0m 0s)' ) )
(0 , 0)
sstr( spoint( '( 0h 2m 30s , 95d 0m 0s)' ) )
(180.625d , 85d)
sstr( spoint( '( 24h 2m 30s , 5d 0m 0s)' ) )
(0.625d , 5d)
sstr( spoint( '( -0h 2m 30s , -5d 0m 0s)' ) )
(359.375d , -5d)
sstr( spoint( '( 0h 2m 30s , -95d 0m 0s)' ) )
(180.625d , -85d)
sstr( spoint( '( 0.625d , -0.04166666667d)' ) )
(0.625d , -0.041666667d)
sstr( spoint( '(0.0109083078249646 , -0.000727220521664407)' ) )
(0.625d , -0.041666667d)
sstr( spoint(0.0109083078249646 , -0.000727220521664407) )
(0.625d , -0.041666667d)
sstr( spoint(7.28318530717958623 , 0.00) )
(1 , 0)
sstr( spoint(0.0 , 2.141592653589793116) )
(3.1415927 , 1)
sdist(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 0m 30s , 0d 0m 0s)' ) )
0.008726646259969247
sdist(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 2m 30s , 10d 0m 0s)' ) )
0.17453292519943317
sdist(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 2m 30s , -10d 0m 0s)' ) )
0.17453292519943253
sdist(spoint( '( 0h 2m 30s , 95d 0m 0s)' ) , spoint( '( 12h 2m 30s , 85d 0m 0s)' ) )
0
sdist(spoint( '( 24h 2m 30s , 10d 0m 0s)' ) , spoint( '( 0h 2m 30s , -10d 0m 0s)' ) )
0.3490658503988658
sdist(spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) )
0
sdist(spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) )
0
spoint_long(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) )
0.01090830782496456
spoint_long(spoint( '( 0h 2m 30s ,95d 0m 0s)' ) )
3.152500961414758
spoint_long(spoint( '( 0h 2m 30s ,85d 0m 0s)' ) )
0.01090830782496456
spoint_long(spoint( '( 0h 2m 30s ,-95d 0m 0s)' ) )
3.152500961414758
spoint_long(spoint( '( 0h 2m 30s ,-85d 0m 0s)' ) )
0.01090830782496456
spoint_long(spoint( '( 0h 2m 30s ,90d 0m 0s)' ) )
0.01090830782496456
spoint_long(spoint( '( 0h 2m 30s ,-90d 0m 0s)' ) )
0.01090830782496456
spoint_long(spoint( '(24h 2m 30s , 0d 0m 0s)' ) )
0.010908307824964325
spoint_long(spoint( '(24h 2m 30s ,95d 0m 0s)' ) )
3.1525009614147574
spoint_long(spoint( '(24h 2m 30s ,85d 0m 0s)' ) )
0.010908307824964325
spoint_long(spoint( '(24h 2m 30s ,-95d 0m 0s)' ) )
3.1525009614147574
spoint_long(spoint( '(24h 2m 30s ,-85d 0m 0s)' ) )
0.010908307824964325
spoint_long(spoint( '(24h 2m 30s ,90d 0m 0s)' ) )
0.010908307824964325
spoint_long(spoint( '(24h 2m 30s ,-90d 0m 0s)' ) )
0.010908307824964325
spoint_lat(spoint( '( 0h 2m 30s , 0d 0m 0s)' ) )
0
spoint_lat(spoint( '( 0h 2m 30s ,95d 0m 0s)' ) )
1.4835298641951802
spoint_lat(spoint( '( 0h 2m 30s ,85d 0m 0s)' ) )
1.4835298641951802
spoint_lat(spoint( '( 0h 2m 30s ,-95d 0m 0s)' ) )
-1.4835298641951802
spoint_lat(spoint( '( 0h 2m 30s ,-85d 0m 0s)' ) )
-1.4835298641951802
spoint_lat(spoint( '( 0h 2m 30s ,90d 0m 0s)' ) )
1.5707963267948966
spoint_lat(spoint( '( 0h 2m 30s ,-90d 0m 0s)' ) )
-1.5707963267948966
spoint_lat(spoint( '(24h 2m 30s , 0d 0m 0s)' ) )
0
spoint_lat(spoint( '(24h 2m 30s ,95d 0m 0s)' ) )
1.4835298641951802
spoint_lat(spoint( '(24h 2m 30s ,85d 0m 0s)' ) )
1.4835298641951802
spoint_lat(spoint( '(24h 2m 30s ,-95d 0m 0s)' ) )
-1.4835298641951802
spoint_lat(spoint( '(24h 2m 30s ,-85d 0m 0s)' ) )
-1.4835298641951802
spoint_lat(spoint( '(24h 2m 30s ,90d 0m 0s)' ) )
1.5707963267948966
spoint_lat(spoint( '(24h 2m 30s ,-90d 0m 0s)' ) )
-1.5707963267948966
spoint_x(spoint( '(0d,-90d)' ) )
6.123233995736766e-17
spoint_y(spoint( '(0d,-90d)' ) )
0
spoint_z(spoint( '(0d,-90d)' ) )
-1
sequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) )
1
sequal( spoint( '( 12h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) )
1
sequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 24h 2m 30s , 90d 0m 0s)' ) )
1
sequal( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) )
1
sequal( spoint( '( 12h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) )
1
sequal( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 24h 2m 30s , -90d 0m 0s)' ) )
1
sequal( spoint( '( 0h 2m 30s , 95d 0m 0s)' ) , spoint( '( 12h 2m 30s , 85d 0m 0s)' ) )
1
sequal( spoint( '( 24h 2m 30s , 10d 0m 0s)' ) , spoint( '( 0h 2m 30s , 10d 0m 0s)' ) )
1
sequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) )
0
sequal( spoint( '( 12h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) )
0
sequal( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 12h 2m 30s , 45d 0m 0s)' ) )
0
snotequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) )
0
snotequal( spoint( '( 12h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) )
0
snotequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 24h 2m 30s , 90d 0m 0s)' ) )
0
snotequal( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) )
0
snotequal( spoint( '( 12h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) )
0
snotequal( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 24h 2m 30s , -90d 0m 0s)' ) )
0
snotequal( spoint( '( 0h 2m 30s , 95d 0m 0s)' ) , spoint( '( 12h 2m 30s , 85d 0m 0s)' ) )
0
snotequal( spoint( '( 24h 2m 30s , 10d 0m 0s)' ) , spoint( '( 0h 2m 30s , 10d 0m 0s)' ) )
0
snotequal( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) )
1
snotequal( spoint( '( 12h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) )
1
snotequal( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 12h 2m 30s , 45d 0m 0s)' ) )
1
sdist( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 0m 30s , 0d 0m 0s)' ) )
0.008726646259969247
sdist( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 2m 30s , 10d 0m 0s)' ) )
0.17453292519943317
sdist( spoint( '( 0h 2m 30s , 0d 0m 0s)' ) , spoint( '( 0h 2m 30s , -10d 0m 0s)' ) )
0.17453292519943253
sdist( spoint( '( 0h 2m 30s , 95d 0m 0s)' ) , spoint( '( 12h 2m 30s , 85d 0m 0s)' ) )
0
sdist( spoint( '( 24h 2m 30s , 10d 0m 0s)' ) , spoint( '( 0h 2m 30s , -10d 0m 0s)' ) )
0.3490658503988658
sdist( spoint( '( 0h 2m 30s , 90d 0m 0s)' ) , spoint( '( 12h 2m 30s , 90d 0m 0s)' ) )
0
sdist( spoint( '( 0h 2m 30s , -90d 0m 0s)' ) , spoint( '( 12h 2m 30s , -90d 0m 0s)' ) )
0
